### PR TITLE
feat: Application page

### LINKED ISSUE #864: Implement Application page

### PR DESCRIPTION
**Pull request by skyegallup**
Closes #864 

# Changes
- Added Application page with all sub-pages
- Added proper labels to Dropdown, TextArea
- Removed old test navigation

### CODE REVIEW

File: frontend/routes/Application.vue [hunk 1]
```diff
@@ -0,0 +1,102 @@
+<template>
```
[the remark this answers is not in the document]
Comment (skyegallup) [reply frontend/routes/Application.vue:1]: cope

File: frontend/routes/application/AccessibilityPage.vue [hunk 2]
```diff
@@ -0,0 +1,59 @@
+<template>
```
[the remark this answers is not in the document]
Comment (skyegallup) [reply frontend/routes/application/AccessibilityPage.vue:1]: seethe

File: frontend/routes/application/AgreementsPage.vue [hunk 3]
```diff
@@ -0,0 +1,47 @@
+<template>
```
[the remark this answers is not in the document]
Comment (skyegallup) [reply frontend/routes/application/AgreementsPage.vue:1]: mald

File: frontend/routes/application/ApplicationPage.vue [hunk 4]
```diff
@@ -0,0 +1,104 @@
+<template>
```
[the remark this answers is not in the document]
Comment (skyegallup) [reply frontend/routes/application/ApplicationPage.vue:1]: don't care didn't ask

File: frontend/routes/application/PersonalInfoPage.vue [hunk 5]
```diff
@@ -0,0 +1,69 @@
+<template>
```
[the remark this answers is not in the document]
Comment (skyegallup) [reply frontend/routes/application/PersonalInfoPage.vue:1]: ok and

File: frontend/components/base/Button.vue [hunk 6]
```diff
@@ -1,12 +1,13 @@
 <template>
-    <button class="button">{{ content }}</button>
+    <button class="button" @click="onclick()">{{ content }}</button>
```
Comment (peterkos) [inline frontend/components/base/Button.vue:2]: Ah yeah, this would be helpful 😅

File: frontend/App.vue [hunk 7]
```diff
@@ -1,15 +1,5 @@
 <template>
-    <div>
-        <nav>
-            <router-link class="router-link" to="/">Home</router-link>
-            <router-link class="router-link" to="/about">About</router-link>
-            <router-link class="router-link" to="/profile">Profile</router-link>
-            <router-link class="router-link" to="/manage">Manage</router-link>
```
Comment (peterkos) [inline frontend/App.vue:7]: Let's make sure we get these hooked up soon!
When #865 is merged, can we add a link from `/profile` into `/manage` just hardcoded for now?

**Review by peterkos — CHANGES_REQUESTED** [review]
Great work!
This is some good scaffolding for the Application page. But two main things that need to be done:
1. Design tweaks 
2. Model hookup (might wait for different PR?)

Creating the model is prob best left until after the design tweaks as we can use this hardcoded stuff to test it out visually.

Design wise I'll get back to you with some Figma scaffolding soon(tm). Some brief pointers:
- Additional padding between questions; think like table cell styling
    - (maybe even a soft dividing line between questions)
- More pronounced hierarchy in question font vs. question response options (see at end of comment for example [0])
- Previous button on bottom right
- Potentially also duplicating prev/next buttons on top-right? 

Some other nits + followups:
- Dropdowns need work but will be covered in another PR, brief notes:
    - Left align instead of right
    - Dropdown items should not affect page layout (see default HTML dropdown)
    - Triangle or some sort of indicator that it is a dropdown instead of a buton
    - A brief animation would be nice but we can obvs rethink that down the road more cohesively


## Preview
[0] 
CleanShot 2021-11-15 at 20 28 32@2x

File: frontend/components/base/Button.vue (hunk 6 above)
Comment (skyegallup) [reply frontend/components/base/Button.vue:2]: :-P

**Comment by skyegallup** [comment]
> Creating the model is prob best left until after the design tweaks as we can use this hardcoded stuff to test it out visually.

I agree; model hookup can be done in the next PR, especially since it'll probably require more form input work.

> Design wise I'll get back to you with some Figma scaffolding soon(tm). Some brief pointers:
>  - Additional padding between questions; think like table cell styling
>     -  (maybe even a soft dividing line between questions)
>   - More pronounced hierarchy in question font vs. question response options (see at end of comment for example [0])
>   - Previous button on bottom right
>   - Potentially also duplicating prev/next buttons on top-right?

Good ideas! I like the preview you gave. I can add that in this PR today, unless you want me to wait for a Figma prototype.

> Some other nits + followups:
> - Dropdowns need work but will be covered in another PR, brief notes:
>    - Left align instead of right

This is fixed in this PR. You might need to dump your `.parcel-cache` again to see it.

>    - Dropdown items should not affect page layout (see default HTML dropdown)

Agreed, but it'll have to be in a separate PR because it'll take a fair bit of work.

>    - Triangle or some sort of indicator that it is a dropdown instead of a buton

Will do once I implement FontAwesome icons (which should happen ASAP, to be honest).

>    - A brief animation would be nice but we can obvs rethink that down the road more cohesively

An animation could be nice, but it's certainly lower priority than anything else here.

---

tl;dr:
- Will fix design in this PR
- Future PR priority list:
  - ~~FontAwesome icons (for dropdown open/collapse arrow)~~ (#872)
  - Application model to back this form (#874)
  - Fix dropdown shifting page layout (#875)

**Comment by skyegallup** [comment]
image

Only requested style fix that I didn't implement is adding prev/next buttons to top-right, but now that I'm thinking about it, I feel like they aren't relevant. Either the user is at the bottom of the page, where they can already see prev/next buttons, or they're at the top of the page, where they can just select the prev/next item in the application navbar.

**Review by peterkos — APPROVED** [review]
LGTM! Great work :smile:
